src/frequenz/sdk/actor/power_distributing/_component_pool_status_tracker.py

```diff
@@ -0,0 +1,163 @@
+# License: MIT
+# Copyright © 2022 Frequenz Energy-as-a-Service GmbH
+
+"""Class that tracks the status of pool of components."""
+
+
+import asyncio
+import logging
+from collections import abc
+
+from frequenz.channels import Broadcast, Receiver, Sender
+from frequenz.channels.util import Merge
+
+from ..._internal._asyncio import cancel_and_await
+from ._component_status import (
+    ComponentPoolStatus,
+    ComponentStatus,
+    ComponentStatusEnum,
+    ComponentStatusTracker,
+    SetPowerResult,
+)
+
+_logger = logging.getLogger(__name__)
+
+
+class ComponentPoolStatusTracker:
```

Comment:
I guess not for this PR, but can't we take the opportunity and include the migration to `BackgroudService` in this big refactoring. These classes handling multiple tasks and having a `stop()` and `join()` look like a perfect candidate for `BackgorundService`.

Reply:
maybe, but right now I'm just doing the bare minimum so I can unblock *pools.

Reply:
I agree for the stuff that was already there and was just renamed/moved, but for new classes I think we should start using `BackgroundService`, it makes no sense to keep piling up technical debt.